datadog_checks_base/datadog_checks/base/utils/db/query.py

```diff
@@ -118,6 +118,7 @@ def compile(self, column_transformers, extra_transformers):
 
         submission_transformers = column_transformers.copy()
         submission_transformers.pop('tag')
+        submission_transformers.pop('tag_list')
```

Comment:
Quick question, but what is the purpose of doing this?

Reply:
Yes, good question as that's not quite obvious from just reading the code.

If I understand correctly the rest of the code here about "extras", and the docs here: https://datadoghq.dev/integrations-core/base/databases/#datadog_checks.base.utils.db.transform.ExtraTransformers

> **Extra transformers**
> Every column transformer (except `tag`) is supported at this level, the only difference being one must set a source to retrieve the desired value.

This variable is used to derive "extra" data, which IIUC should all correspond to metric-submittable values.

Indeed as we can see here:

https://github.com/DataDog/integrations-core/blob/b9808db30883fa17bb81be60df2c1c5547e0ef3c/datadog_checks_base%2Fdatadog_checks%2Fbase%2Futils%2Fdb%2Fcore.py#L139-L144

Column values go in the `submission_queue` to be submitted (eg with gauge, monotonic_count, etc), _except_ these `tag` and `tag_list` values which are used to add tags, rather than submit specific metrics.

Reply:
Gotcha so `tag` and `tag_list` are excluded from being submitted as specific metrics in the end. 

In that case, what is the point of having them in `column_transformers` in the first place?

Reply:
@yzhan289 

> In that case, what is the point of having them in column_transformers in the first place?

1/ Transformer gets computed from `column_transformers`. In the context of this PR this calls `get_tag_list()` which returns the `tag_list` callable.

https://github.com/DataDog/integrations-core/blob/b9808db30883fa17bb81be60df2c1c5547e0ef3c/datadog_checks_base%2Fdatadog_checks%2Fbase%2Futils%2Fdb%2Fquery.py#L100

2/ Column name, column_type (`tag_list`) and transformer (`tag_list()`) get put into `column_data`.

https://github.com/DataDog/integrations-core/blob/b9808db30883fa17bb81be60df2c1c5547e0ef3c/datadog_checks_base%2Fdatadog_checks%2Fbase%2Futils%2Fdb%2Fquery.py#L112-L113

3/ `column_data` is eventually iterated over in `compile_queries()`, leading to the if/elif/else block I linked to earlier that actually does the thing we want: `tags.extend()`.

It's "_highly dynamic_" as per @ofek's own words, but this PR proves it's also fairly easily extensible at the same time. :-)

Reply:
Thanks! Very clear now.